Shift3/boilerplate-client-react
# Fix toast notification styling

## Changes
1. Implement notification service using `react-tostify` service.
2. Match styling of toast notifications to Angular boilerplate.
3. Remove old implementation of toast notification using redux.
4. Refactor components and tests to use the new notification service.

## Purpose
The styling of the toast notifications needed to be updated to match the toast notifications in the Angular boilerplate. In addition, our custom implementation of toast notifications was very basic. The `react-tostify` package supports many features that will make customization in future derivative projects much easier. The Angular boilerplate also uses a package for toasts rather than implementing a custom module.

## Approach
- Calls to `react-toastify` package are wrapped in a light-weight service to keep components de-coupled from implementation details. As a bonus, since the service is just simple functions instead of hooks like before, the service methods can be used anywhere in the app and not just inside components.
- A custom styled `NotificationContainer` is created which wraps the `ToastContainer` provided by the package. Default configurations for toast behavior are set as attributes here.

## Learning
[react-tostify](https://www.npmjs.com/package/react-toastify) is a mature/stable toast notification package that has many features and allows for a lot of customization. Check out its [docs](https://fkhadra.github.io/react-toastify/introduction)

## Testing
1. Pull in the changes to your local copy of this branch.
2. Run `yarn install` to install the new dependencies.
3. Run `yarn start` to start the application.
4. Use the application (e.g. create an account, reset password, login and try to create a user) and check that the new toast notifications are shown.

### Closes #442
